test/kira/accounts/queries/find_free_developer_test.exs

```diff
@@ -0,0 +1,79 @@
+defmodule KiraTest.Accounts.Queries.FindFreeDeveloperTest do
+  use Kira.DataCase
+
+  import KiraTest.Factory
+  alias Kira.Accounts.Queries.FindFreeDeveloper
+
+  describe "find task to assign service" do
+    setup do
+      {:ok, project: insert(:project)}
+    end
+
+    test "with no users on project", %{project: project} do
+      {:ok, context} = FindFreeDeveloper.run(project_uid: project.uid)
+
+      assert context.users == []
+    end
+
+    test "with a user and without issues", %{project: project} do
+      user = insert(:user, projects: [project])
+
+      {:ok, context} = FindFreeDeveloper.run(project_uid: project.uid)
+
+      user_ids = context.users |> Enum.map(fn u -> u.id end)
+
+      assert user_ids == [user.id]
+    end
+
+    test "with a user and unassigned issues", %{project: project} do
+      user = insert(:user, projects: [project])
+      insert_list(3, :issue, %{project: project, assignee: nil})
+
+      {:ok, context} = FindFreeDeveloper.run(project_uid: project.uid)
+
+      user_ids = context.users |> Enum.map(fn u -> u.id end)
+
+      assert user_ids == [user.id]
```

Comment:
@sobolevn this one fails, if you could point the problem within `UserQueries` or maybe said what's a better approach there, it would help. Thanks

Reply:
Can you please show the output?

Reply:
It's pretty simple:
```
Assertion with == failed
     code:  assert user_ids == [user.id()]
     left:  []
     right: [5135]
```

So something is wrong with the queries `assigned_within_project` and `not_assigned_within_project` at `lib/kira/accounts/queries/user.ex`. I'm trying to fix it, but there's a chance you'll see the problem just quicker due to the experience

Reply:
My wild guess is that it happens because of the double `from u in for_project(query, project),` https://github.com/wemake-services/kira/blob/6e7e054a0231ba7e80728f5f4e0e9a8bf5495119/lib/kira/accounts/queries/user.ex#L36